docs/mkdocs/docs/examples/nlohmann_json_deserialize_enum_strict.cpp

```diff
@@ -0,0 +1,57 @@
+#include <iostream>
+#include <nlohmann/json.hpp>
+
+#ifdef __cpp_exceptions
+    #undef __cpp_exceptions
+    #define __cpp_exceptions 1
+#endif
+
+#ifdef JSON_NOEXCEPTION
+    #define JSON_NOEXCEPTION 0
+#endif
+
+#if (defined(__cpp_exceptions) || defined(__EXCEPTIONS) || defined(_CPPUNWIND)) && !defined(JSON_NOEXCEPTION)
```

Comment:
these lines 13 to 24 are copied from json.hpp; this is required as a workaround to get through the following compilation issues
![Screenshot from 2025-01-20 00-13-05](https://github.com/user-attachments/assets/e0116c6d-bb82-4dcc-bff5-0420d761a82e)
@nlohmann  @gregmarr @codenut please advise the proper way to resolve this compilation issue.

Reply:
Compiler: c++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0;